junit-platform-commons/src/main/java/org/junit/platform/commons/util/ReflectionUtils.java

```diff
@@ -725,7 +732,27 @@ private static boolean isMethodShadowedBy(Method upper, Method lower) {
 				return false;
 			}
 		}
-		return true;
+		// lower is sub-signature of upper: check for generics in upper method
+		if (isGeneric(upper)) {
```

Comment:
At a glance, it appears that this _may_ suffice, especially if the tests are all passing now. However, I can't be certain that it covers all use cases without a thorough investigation on my part. In any case, it appears that it might well be "_good enough_" for the time being.